src/bytes/src/utils.cairo

```diff
@@ -1,7 +1,63 @@
+use alexandria_bytes::{Bytes, BytesTrait};
 use alexandria_data_structures::array_ext::ArrayTraitExt;
+use core::fmt::{Debug, Display, Formatter, Error};
+use core::to_byte_array::{AppendFormattedToByteArray, FormatAsByteArray};
 use integer::u128_byte_reverse;
 use keccak::{u128_to_u64, u128_split as u128_split_to_u64, cairo_keccak};
 
+fn format_byte_hex(byte: u8, ref f: Formatter) -> Result<(), Error> {
+    let base: NonZero<u8> = 16_u8.try_into().unwrap();
+    if byte < 0x10 {
+        // Add leading zero for single digit numbers
+        let zero: ByteArray = "0";
+        let res = Display::fmt(@zero, ref f);
+        if res.is_err() {
+            return res;
+        }
+    }
+    Display::fmt(@byte.format_as_byte_array(base), ref f)
+}
+
+impl BytesDebug of Debug<Bytes> {
+    fn fmt(self: @Bytes, ref f: Formatter) -> Result<(), Error> {
+        let mut i: usize = 0;
+        let prefix: ByteArray = "0x";
+        let mut res = Display::fmt(@prefix, ref f);
+        if res.is_err() {
+            return res;
+        }
+        while i < self.size() {
+            let (new_i, value) = self.read_u8(i);
+            res = format_byte_hex(value, ref f);
+            if res.is_err() {
+                break;
```

Comment:
If res is an error do you really want to return the last correct res or do you also want to propagate the error ?

Reply:
I think propagating the error would be ideal. Otherwise, the user gets incomplete hex representations that cause confusion.
Is this the best way to do that, since you cannot use `?` in a loop?

Reply:
ah yes i always forget

Reply:
so return the error instead of just breaking ? or `break res` and propagate the error after